### PR TITLE
test: enable workload identity test with soak clusters 

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Enabling workload identity with soak clusters
  - Updated the soak clusters with `--enable-oidc-issuer`.
  - Setup federated identity credential with the issuer and <service account name, namespace>

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by aramase** [comment]
Test run: https://dev.azure.com/AzureContainerUpstream/Secrets%20Store%20CSI%20Driver%20Provider%20Azure/_build/results?buildId=39093&view=results

File: test/e2e/workload_identity_test.go [hunk 1]
```diff
@@ -103,8 +103,8 @@ var _ = Describe("CSI inline volume test with workload identity", func() {
 	})
 
 	It("should read secret, key from pod", func() {
-		if !config.IsKindCluster {
-			Skip("test case currently supported for kind cluster only")
+		if !(config.IsKindCluster || config.IsSoakTest) {
```
Comment (nilekhc) [inline test/e2e/workload_identity_test.go:106]: For now could we skip this test for Arc? The clusters created for conformance testing are not setup for WI, hence this test is failing.
We set `IsKindCluster=True` and `IsArcTest=true` while running conformance test

**Review by nilekhc — APPROVED** [review]

**Pull request merged**